src_tests/test_lossy_boson_sampling_simulator.py

```diff
@@ -0,0 +1,81 @@
+__author__ = 'Tomasz Rybotycki'
+
+import unittest
+
+from math import factorial
+from numpy import zeros, array
+from typing import List
+from src.BosonSamplingSimulator import BosonSamplingSimulator
+from src.simulation_strategies.FixedLossSimulationStrategy import FixedLossSimulationStrategy
+from src.ExactLossyBosonSamplingDistributionCalculator.ExactLossyBosonSamplingDistributionCalculator import ExactLossyBosonSamplingDistributionCalculator,\
+    BosonSamplingExperimentConfiguration
+from src.Quantum_Computations_Utilities import calculate_total_variation_distance
+
+
+class TestClassicalLossyBosonSamplingSimulator(unittest.TestCase):
+
+    def setUp(self) -> None:
+        # Generate permutation matrix and define initial state.
+        self.initial_state = [1, 1, 1, 0, 0]
+        self.permutation_matrix = array([
+            [0, 0, 1, 0, 0],
+            [1, 0, 0, 0, 0],
+            [0, 0, 0, 1, 0],
+            [0, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0],
+        ])
+
+        # Define some additional variables for more clear experiment configuration assignment.
+        self.number_of_particles_lost = 2
+
+        # Create configuration object.
+        self.experiment_configuration = BosonSamplingExperimentConfiguration(
+            interferometer_matrix=self.permutation_matrix,
+            possible_outcomes=[[1, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 0, 0, 1]],
+            initial_state=self.initial_state,
+            initial_number_of_particles=sum(self.initial_state),
+            number_of_modes=len(self.initial_state),
+            number_of_particles_lost=self.number_of_particles_lost,
+            number_of_particles_left=sum(self.initial_state) - self.number_of_particles_lost
+        )
+
+    def test_approximate_and_exact_distribution_distance(self) -> None:
+        exact_distribution_calculator = ExactLossyBosonSamplingDistributionCalculator(self.experiment_configuration)
+        exact_distribution = exact_distribution_calculator.calculate_exact_distribution()
+        approximate_distribution = self.__calculate_approximate_distribution()
+        distance = calculate_total_variation_distance(exact_distribution, approximate_distribution)
+        bound = self.__calculate_distribution_error_bound()
+        self.assertTrue(distance < bound)
```

Comment:
Consider using `assertGreater`/`assertLess` (https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertGreater), they give more information in case of an error.

Reply:
+1